sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricKey.java

```diff
@@ -36,12 +37,11 @@ public abstract class MetricKey implements Serializable {
   public abstract MetricName metricName();
 
   @Override
-  @Memoized
   public String toString() {
     return stepName() + ":" + metricName();
   }
 
   public static MetricKey create(@Nullable String stepName, MetricName metricName) {
-    return new AutoValue_MetricKey(stepName, metricName);
+    return new CachedToStringMetricKey(new AutoValue_MetricKey(stepName, metricName));
```

Comment:
ha, I thought that we will create a `CachedToStringMetricKey` , that replaces the AutoValue_MetricKey instead of wrapping the AutoValue_MetricKey.  But either way should be fine.

Reply:
Not a fan of autovalue.  :)

Reply:
Haha, I see what you mean! Initially, I also thought about directly creating a CachedToStringMetricKey to replace AutoValue_MetricKey, but I realized that AutoValue provides its own implementations of .equals and .hashcode, which I wanted to preserve. That’s why it evolved into the wrapper approach—my goal was to minimize changes while keeping the core behavior intact.